nncf/experimental/tensor/functions.py

```diff
@@ -324,28 +322,105 @@ def zeros_like(a: TTensor) -> TTensor:
     return Tensor(zeros_like(a.data))
 
 
-__all__ = [
-    "device",
-    "squeeze",
-    "flatten",
-    "max",
-    "min",
-    "abs",
-    "astype",
-    "reshape",
-    "all",
-    "allclose",
-    "any",
-    "count_nonzero",
-    "isempty",
-    "isclose",
-    "maximum",
-    "minimum",
-    "ones_like",
-    "minimum",
-    "where",
-    "zeros_like",
-]
+@functools.singledispatch
+def stack(x: List[TTensor], axis: int = 0) -> TTensor:
+    """
+    Stacks a list or deque of Tensors rank-R tensors into one Tensor rank-(R+1) tensor.
+
+    :param x: List or deque of Tensors.
+    :param axis: The axis to stack along.
+    :return: Stacked Tensor.
+    """
+    if isinstance(x, List):
+        unwrapped_x = [i.data for i in x]
+        # singledispatch cannot dispatch function by element in a list
```

Comment:
Same limitation of the `singledispatch` mechanism necessitates complications such as this. I imagine that you will run into this yet, for example, when implementing an equivalent of `np.concatenate` required [here](https://github.com/openvinotoolkit/nncf/blob/c07810beeb2a638ba8c1d1800167f2f0bb97cabe/nncf/common/tensor_statistics/reduction.py#L57), so might as well pull the logic here out into a helper function.

Notice that the OOP approach allowed to immediately write such functions ([interface](https://github.com/openvinotoolkit/nncf/pull/2032/files#diff-ec72fc8ffe89978a911fbca7e7f6a1bc30ba838d79b3b4b59a43d3b81aec910fR223), [implementation](https://github.com/openvinotoolkit/nncf/pull/2032/files#diff-a4ab3aa944f7ba33bcea038694de8d08784f5124668ba7a08e585fc3e47e5ff6R109))

Reply:
Added function `_dispatch_list`